lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -581,10 +594,11 @@ def create_scsi_controller(self, scsi_type):
         scsi_ctl.operation = vim.vm.device.VirtualDeviceSpec.Operation.add
         scsi_device = self.scsi_device_type.get(scsi_type, vim.vm.device.ParaVirtualSCSIController)
         scsi_ctl.device = scsi_device()
-        scsi_ctl.device.busNumber = 0
+        # busNumber value of scsi controllers increases
+        # scsi_ctl.device.busNumber = 0
         # While creating a new SCSI controller, temporary key value
         # should be unique negative integers
-        scsi_ctl.device.key = -randint(1000, 9999)
+        scsi_ctl.device.key = -randint(1000, 1999)
```

Comment:
This increases the chance of collision and should be changed back to 9999.

Reply:
I noticed from the manually deployed VM, different type of hardware has the default range of key, e.g., SCSI controller starts from 1000, the following added will add 1 from it, disk starts from 2000, network adapter starts from 4000, don't know if randomly generated large range number will affects other devices. Thanks.

Reply:
I think these `randint` calls should be removed entirely, all the increase in the `randint` range does is reduce the chance of collision not remove it. Some poor user will get their build failing randomly because of collision, which is a risk that increases with every device the user adds. The docs suggest that it just needs to be a unique negative number which will get replaced when the device is actually created.

I suggest putting a `device_key` attribute on the `PyVmomiHelper` object, with a getter function that on each call will decrement and return a negative integer (.e.g. -1,-2,-3). This would remove any chance of collision, and fulfil the requirement of giving every new device a unique, negative temporary id.